js/src/components/google-combo-account-card/connected-google-combo-account-card.js

```diff
@@ -69,27 +81,87 @@ const ConnectedGoogleComboAccountCard = () => {
 		upsertAccount();
 	}, [ finalizeAdsAccountCreation, upsertAdsAccount, invalidateResolution ] );
 
-	if ( ! hasDetermined ) {
-		return <SpinnerCard />;
-	}
+	const handleCancelClick = () => {
+		setEditMode( false );
+	};
 
-	// @TODO: edit mode implementation in 2605
-	const editMode = false;
-	const shouldClaimGoogleAdsAccount = Boolean(
-		! loading && googleAdsAccount?.id && hasAccess === false
+	const handleEditClick = () => {
+		setEditMode( true );
+	};
+
+	// During MC account creation, we need to show the ConnectMC component
+	// when the account creation needs to show the Switch or Reclaim flow.
+	const googleMCHasError = [ 409, 403 ].includes(
+		resultCreateMCAccount.response?.status
 	);
 
-	const hasExistingGoogleMCAccounts = existingGoogleMCAccounts?.length > 0;
+	// After creating a new account, it may be connected but not ready
+	// (e.g., needing to reclaim the URL). In this case, we show the ConnectMC
+	// component, even if the existing accounts list has not yet updated.
 	const showConnectMC =
-		( editMode && hasExistingGoogleMCAccounts ) ||
-		( ! isGoogleMCReady && hasExistingGoogleMCAccounts );
-
-	const hasExistingGoogleAdsAccounts = existingGoogleAdsAccounts?.length > 0;
+		( googleMCHasError ||
+			hasGoogleMCConnection ||
+			hasExistingGoogleMCAccounts ) &&
+		( editMode || ! isGoogleMCReady );
+
+	// After creating a new account, it may not show up in the existing accounts list
+	// immediately. In this case, we show the ConnectAds component in edit mode unless
+	// we're showing the claim notice in the upper card.
 	const showConnectAds =
-		( ( editMode && hasExistingGoogleAdsAccounts ) ||
-			( ! isConnected && hasExistingGoogleAdsAccounts ) ) &&
+		( isConnected || hasExistingGoogleAdsAccounts ) &&
```

Comment:
The `isConnected ||` seems to be redundant?

Reply:
Similar to showConnectMC case above, A new Ads account might be connected but not yet returning in the existing accounts list. In that case, it should be shown when we enter editMode.

Reply:
In the video, the `isConnected ||` condition at line 111 was commented out. After the new Google Ads account was automatically created and claimed. The newly created account was shown in the existing accounts. Not sure if I was missing anything?

https://github.com/user-attachments/assets/1f564317-f993-427f-a863-fdd1b2567942